src/state/apis/covalent/validators.ts

```diff
@@ -0,0 +1,264 @@
+      owner: z.string().nullable(),
+    })
+    .nullable(),
+  owner: z.string().nullable(),
+  owner_address: z.string().nullable(),
+  burned: z.boolean().nullable(),
+})
+
+const NftCollectionSchema = z.object({
+  contract_decimals: z.number().nullable(),
+  contract_name: z.string(),
+  contract_ticker_symbol: z.string(),
+  contract_address: z.string(),
+  supports_erc: z.array(z.string().nullable()).nullable(),
+  logo_url: z.string().nullable(),
+  last_transferred_at: z.string(),
+  native_token: z.boolean(),
+  type: z.literal('nft'),
+  balance: z.string(),
+  balance_24h: z.string(),
+  quote_rate: z.string().nullable(),
+  quote_rate_24h: z.string().nullable(),
+  quote: z.string().nullable(),
+  pretty_quote: z.string().nullable(),
+  quote_24h: z.string().nullable(),
+  pretty_quote_24h: z.string().nullable(),
+  nft_data: z.array(NftItemSchema),
+  is_spam: z.boolean().nullable(),
+})
+
+const CovalentNftItemSchema = z.object({
+  contract_decimals: z.number().nullable(),
+  contract_name: z.string(),
+  contract_ticker_symbol: z.string(),
+  contract_address: z.string(),
+  supports_erc: z.array(z.string()).nullable(),
+  logo_url: z.string().nullable(),
+  last_transferred_at: z.string(),
+  native_token: z.boolean().nullable(),
+  type: z.string(),
+  balance: z.string(),
+  balance_24h: z.string(),
+  quote_rate: z.string().nullable(),
+  quote_rate_24h: z.string().nullable(),
+  quote: z.string().nullable(),
+  pretty_quote: z.string().nullable(),
+  quote_24h: z.string().nullable(),
+  pretty_quote_24h: z.string().nullable(),
+  nft_data: z.array(NftItemSchema).nullable(),
+  is_spam: z.boolean().nullable(),
+})
+
+const CovalentNftUserTokensResponseSchema = z.object({
+  data: z.object({
+    address: z.string(),
+    updated_at: z.string(),
+    next_update_at: z.string(),
+    quote_currency: z.string(),
+    chain_id: z.number(),
+    chain_name: z.string(),
+    items: z.array(CovalentNftItemSchema),
+    is_spam: z.boolean().nullable(),
+  }),
+  error: z.boolean(),
+  error_message: z.string().nullable(),
+  error_code: z.number().nullable(),
+})
+
+export type CovalentNftUserTokensResponseType = Infer<typeof CovalentNftUserTokensResponseSchema>
+export type CovalentNftItemSchemaType = Infer<typeof CovalentNftItemSchema>
+export type CovalentNftCollectionType = Infer<typeof NftCollectionSchema>
+
+export const parseToV2NftUserItem = (
+  covalentItem: CovalentNftItemSchemaType,
+  chainId: ChainId,
+): V2NftUserItem => {
+  const medias = covalentItem.nft_data?.[0]?.external_data?.image
+    ? [
+        {
+          type: 'image',
+          originalUrl: covalentItem.nft_data[0].external_data.image,
+        },
+      ]
+    : []
+  const token: V2ZapperNft = {
+    id: covalentItem.contract_address,
+    name: covalentItem.contract_name,
+    tokenId: covalentItem.nft_data![0].token_id,
+    lastSaleEth: null, // Covalent doesn't give us this
```

Comment:
Actually a native asset (MATIC), not ETH, but conforms to the `V2ZapperNft` type